java-tools/OsmAndMapCreatorUtilities/src/main/java/net/osmand/routes/RouteRelationExtractor.java

```diff
@@ -297,7 +297,26 @@ private void saveGpx(Entity relation, Map<EntityId, Entity> children, File resul
 //		DEBUG = relation.getId() == 13168625; // TODO remove debug
 
 		GPXFile gpxFile = new GPXFile(OSMAND_ROUTER_V2);
-		gpxFile.metadata.name = Objects.requireNonNullElse(relation.getTag("name"), String.valueOf(relation.getId()));
+
+		String id = String.valueOf(relation.getId());
+		String ref = relation.getTag("ref");
+		String mainName = relation.getTag("name");
+		String enName = relation.getTag("name:en");
+		String description = relation.getTag("description");
+		final int MAX_DESC_NAME_LENGTH = 80;
```

Comment:
not needed osm tags always < 255

Reply:
There are some cases of "unnamed" relation with description > 80 chars:

https://www.openstreetmap.org/relation/11817818
https://www.openstreetmap.org/relation/8103090
https://www.openstreetmap.org/relation/16137765
https://www.openstreetmap.org/relation/3237876

I'd like to limit the length of description-as-name to avoid smth like this: "Účelová komunikácia vyhradená pre individuálny vjazd s bicyklom na územie NAPANT - oficiálne podľa návštevného poriadku až na sedlo pod Kúpeľom, ale keďže to nie je moc cyklistický terén, tak neznačím)" as a name...

There are a much more cases when description-as-name would be OK.

Reply:
unlimited